### PR TITLE
fix for issue #5832

### LINKED ISSUE #5832: vpn-seed-server pods fail to start in a local garden kind cluster
/area dev-productivity
/area testing
/kind bug

**What happened**:
Tried to setup a local garden cluster following the instructions [here](https://github.com/gardener/gardener/blob/master/docs/deployment/getting_started_locally.md).
* Gardener components successfully started
* Gardenlet successfully bootstrapped the seed
* Shoot cluster creation failed reconciliation and was always stuck at 80%. On closer inspection 2 of the pods never came up.
   * machine-shoot--local--local-local
   * vpn-seed-server

**What you expected to happen**:
Shoot cluster should have been successfully reconciled and started.
> Question: Is https://github.com/gardener/vpn2 built for ARM64?

**How to reproduce it (as minimally and precisely as possible)**:
Just run the gardener local setup using the instructions at [](https://github.com/gardener/gardener/blob/master/docs/deployment/getting_started_locally.md) and run this on the latest macbook with GOARCH = arm64.

**Anything else we need to know?**:
**Controller registrations were properly setup**
```
k get controllerregistration                                          
NAME                RESOURCES                                                                                                                                                        AGE
networking-calico   Network/calico                                                                                                                                                   42m
provider-local      BackupBucket/local, BackupEntry/local, DNSProvider/local, 
```
**ControllerInstallations are fine as well**
```
k get controllerinstallations                                         
NAME                      REGISTRATION        SEED    VALID   INSTALLED   HEALTHY   AGE
networking-calico-xwsnk   networking-calico   local   True    True        True      42m
provider-local-x2k2t      provider-local      local   True    True        True      42m
```
**Pod status in shoot namespace:**
```
kubectl get po -n shoot--local--local
NAME                                            READY   STATUS             RESTARTS   AGE
etcd-events-0                                   2/2     Running            0          10h
etcd-main-0                                     2/2     Running            0          10h
gardener-resource-manager-6d8b4c8c5f-4jp8h      1/1     Running            0          10h
gardener-resource-manager-6d8b4c8c5f-cs68j      1/1     Running            0          10h
gardener-resource-manager-6d8b4c8c5f-rggch      1/1     Running            0          10h
grafana-operators-5b6f77f499-x5nhk              1/1     Running            0          10h
grafana-users-c445495fc-h59hj                   1/1     Running            0          10h
kube-apiserver-768c8f7779-5sx2z                 2/2     Running            0          10h
kube-controller-manager-7758c4b9fc-78442        1/1     Running            0          10h
kube-scheduler-7f7d5bdb8d-xvd59                 1/1     Running            0          10h
machine-controller-manager-6d7f6ff797-5wllt     2/2     Running            0          10h
machine-shoot--local--local-local-58d9b-5vklt   0/1     Running            0          16m
vpn-seed-server-7c9888fd69-8zlbk                1/2     CrashLoopBackOff   126        10h
```
**Logs from vpn-seed-server-7c9888fd69-8zlbk => container vpn-seed-server**
```
2022-04-21 03:30:36 WARNING: --topology net30 support for server configs with IPv4 pools will be removed in a future release. Please migrate to --topology subnet as soon as possible.
2022-04-21 03:30:36 DEPRECATED OPTION: --cipher set to 'AES-256-CBC' but missing in --data-ciphers (AES-256-GCM:AES-128-GCM). Future OpenVPN version will ignore --cipher for cipher negotiations. Add 'AES-256-CBC' to --data-ciphers or change --cipher 'AES-256-CBC' to --data-ciphers-fallback 'AES-256-CBC' to silence this warning.
2022-04-21 03:30:36 WARNING: file '/srv/secrets/vpn-server/tls.key' is group or others accessible
2022-04-21 03:30:36 OpenVPN 2.5.2 x86_64-alpine-linux-musl [SSL (OpenSSL)] [LZO] [LZ4] [EPOLL] [MH/PKTINFO] [AEAD] built on May  4 2021
2022-04-21 03:30:36 library versions: OpenSSL 1.1.1k  25 Mar 2021, LZO 2.10
2022-04-21 03:30:36 NOTE: the current --script-security setting may allow this configuration to call user-defined scripts
2022-04-21 03:30:36 TUN/TAP device tun0 opened
2022-04-21 03:30:36 /sbin/ip link set dev tun0 up mtu 1500
2022-04-21 03:30:36 /sbin/ip link set dev tun0 up
2022-04-21 03:30:36 /sbin/ip addr add dev tun0 local 192.168.123.1 peer 192.168.123.2
iptables v1.8.6 (legacy): can't initialize iptables table `filter': iptables who? (do you need to insmod?)
Perhaps iptables or your kernel needs to be upgraded.
iptables v1.8.6 (legacy): can't initialize iptables table `filter': iptables who? (do you need to insmod?)
Perhaps iptables or your kernel needs to be upgraded.
2022-04-21 03:30:36 /firewall.sh on tun0 1500 1623 192.168.123.1 192.168.123.2 init
2022-04-21 03:30:36 WARNING: Failed running command (--up/--down): external program exited with error status: 3
2022-04-21 03:30:36 Exiting due to fatal error
```

**Environment**:
GOOS = darwin
GOARCH = arm64
MacOs version: Monterey 12.3.1

- Gardener version: 
- Kubernetes version (use `kubectl version`):
Client Version: version.Info{Major:"1", Minor:"23", GitVersion:"v1.23.5", GitCommit:"c285e781331a3785a7f436042c65c5641ce8a9e9", GitTreeState:"clean", BuildDate:"2022-03-16T15:51:05Z", GoVersion:"go1.17.8", Compiler:"gc", Platform:"darwin/arm64"}
Server Version: version.Info{Major:"1", Minor:"21", GitVersion:"v1.21.1", GitCommit:"5e58841cce77d4bc13713ad2b91fa0d961e69192", GitTreeState:"clean", BuildDate:"2021-06-22T04:50:02Z", GoVersion:"go1.16.4", Compiler:"gc", Platform:"linux/arm64"}
WARNING: version difference between client (1.23) and server (1.21) exceeds the supported minor version skew of +/-1
- Cloud provider or hardware configuration:
- Others:

### PR DESCRIPTION
**Pull request by unmarshall**
**How to categorize this PR?**
/area dev-productivity
/area testing
/kind bug

**What this PR does / why we need it**:
Fixes the issues of starting local gardener cluster with seed/shoot on ARM64 macos devices

**Which issue(s) this PR fixes**:
Fixes #5832 

**Special notes for your reviewer**:
Solves the pending issues for starting shoot machine on:
GOOS = darwin
GOARCH = arm64
MacOs version: Monterey 12.3.1

### CODE REVIEW

File: example/provider-local/base/cloudprofile.yaml [hunk 1]
```diff
@@ -8,6 +8,7 @@ spec:
   - name: local
   kubernetes:
     versions:
+    - version: 1.23.6
     - version: 1.23.1
```
Comment (rfranzke) [inline example/provider-local/base/cloudprofile.yaml]: You could drop this line now

Comment (unmarshall) [reply example/provider-local/base/cloudprofile.yaml]: done

**Review by rfranzke — COMMENTED** [review]
/lgtm
/approve

File: example/provider-local/base/cloudprofile.yaml (hunk 1 above)
Comment (acumino) [reply example/provider-local/base/cloudprofile.yaml]: Please make the changes in the e2e tests also.

**Review by acumino — COMMENTED** [review]
/lgtm

**Pull request merged**